language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/OrderByClauseNodeContext.java

```diff
@@ -93,18 +94,22 @@ public void sort(BallerinaCompletionContext context,
                 TypeDescKind.BOOLEAN, TypeDescKind.FLOAT,
                 TypeDescKind.DECIMAL);
 
+        QueryExpressionNode queryExprNode = (QueryExpressionNode) node.parent().parent();
+        List<String> filteredSymbolNames = SortingUtil.getVisibleSymbolNamesWithinNodeAndCursor(context, queryExprNode);
```

Comment:
Here we get a list of names. For that, we have to iterate through the visible symbols. Since this is valid for the symbol completion items, what if we do the following.
Within our util, we add a validation only, when given the symbol and the node. Then we do not need to iterate over the visible symbols again. For each symbol completion item, we just validate it.

Reply:
@nadeeshaan  You mean to not get the filtered visible symbols prior and loop through the visible symbols using `anymatch` inside the `forEach` of the completion Items?. In this implementation, I only iterate through visible symbols once and use that list.

Reply:
Removed the iteration of visible symbols